charts/openstack-cluster/templates/control-plane/kubeadm-control-plane.yaml

```diff
@@ -25,6 +25,14 @@ preKubeadmCommands:
 {{- end }}
 {{- end }}
 
+{{- define "openstack-cluster.controlplane.kubeadmConfigSpec.containerdRestart" -}}
+{{- if and .Values.proxy.http_proxy .Values.proxy.https_proxy .Values.proxy.no_proxy }}
+preKubeadmCommands:
+  - systemctl daemon-reload
+  - systemctl restart containerd
+{{- end }}
+{{- end }}
+
```

Comment:
This can be added to the `openstack-cluster.kubeadmConfigSpec.containerd` template in `_helpers.tpl`.

Also, did you test that this is definitely required? It might be possible that cloud-init creates the drop-in file early enough that we don't need the `daemon-reload+restart`.